cls/SourceControl/Git/Utils.cls

```diff
@@ -1533,6 +1630,29 @@ ClassMethod RunGitCommandWithInput(command As %String, inFile As %String = "", O
     quit returnCode
 }
 
+ClassMethod GenerateCommitMessageFromFiles(filesWithActions) As %String
+{
+    set commitMsg = ""
+
+    set iterator = filesWithActions.%GetIterator()
+    while iterator.%GetNext(,.value,) {
+        set action = value.%Get("action")
+        set file = value.%Get("file")
+
+
+        if action = "A" {
+            set commitMsg = commitMsg _ "added " _ file _ ", "
+        } elseif action = "M" {
+            set commitMsg = commitMsg _ "modified " _ file _ ", "
+        } elseif action = "D" {
+            set commitMsg = commitMsg _ "deleted " _ file _ ", "
+        } elseif action = "R" {
+            set commitMsg = commitMsg _ "renamed " _ file _ ", "
+        }
+    }
+    quit commitMsg
```

Comment:
Useful pattern for things like this to avoid the trailing ", " that you'll currently have:
set commitMsg = ""

In loop:
set commitMsg = commitMsg_$ListBuild(moreText)

At end:
quit $ListToString(commitMsg,", ")